### ISSUE TITLE
Solidity Test that can run multiple solidity versions at once.

### ISSUE THREAD

**Issue by karmabadger** [foundry-rs/foundry#3058]
### Component

Forge

### Describe the feature you would like

Solidity Test that can run multiple solidity versions at once. From time to time, you might need to test new code against/with old code such as UniswapV2 which require completely incompatible versions. Currently this is not possible to do as it wouldn't even compile.
(is there even a work around)?

**Comment by 0xPhaze** [comment]
A somewhat cumbersome workaround is to compile it and to `vm.etch` the runtime code. You can also manually deploy it (with some args) and copy the code to etch (see https://github.com/0xPhaze/paradigm-ctf-2021-foundry/blob/master/src/ParadigmSetup.sol as an example). Or you can get the creation code and concatenate with your constructorArgs.

**Comment by rkrasiuk** [comment]
related #2704

**Comment by karmabadger** [comment]
> A somewhat cumbersome workaround is to compile it and to `vm.etch` the runtime code. You can also manually deploy it (with some args) and copy the code to etch (see https://github.com/0xPhaze/paradigm-ctf-2021-foundry/blob/master/src/ParadigmSetup.sol as an example). Or you can get the creation code and concatenate with your constructorArgs.

ok I'll try that. Ultimately the best way to fix this would a fix for backwards compatibility in solc directly. but yeah anything is appreciated rn

**Comment by mds1** [comment]
Just noting this is related to #3062 which proposes a config format for supporting multiple solc versions/settings

**Comment by mds1** [comment]
Oh maybe I misunderstood then. I interpreted the issue as being for different test files. You want to have two different solc versions uses in the same file? If solc doesn't support that, it seems more like a solidity issue than a foundry issue IIUC

**Comment by karmabadger** [comment]
> Oh maybe I misunderstood then. I interpreted the issue as being for different test files. You want to have two different solc versions uses in the same file? If solc doesn't support that, it seems more like a solidity issue than a foundry issue IIUC

yes it is preferrably fixed with solc but there should be a good supported work around tbh. This is a pretty common issue imo.

**Comment by fubhy** [comment]
Would it be feasible to come up with a cheatcode that produces sth. like [0xPhaze/paradigm-ctf-2021-foundry@master/src/ParadigmSetup.sol](https://github.com/0xPhaze/paradigm-ctf-2021-foundry/blob/master/src/ParadigmSetup.sol?rgh-link-date=2022-09-02T08%3A35%3A38Z) without that cumbersome manual initialization?

**Comment by stevyhacker** [comment]
I'd like the ability to have multiple solc versions configured like Hardhat allows to do with the `compilers` filed in hardhat.config
This is closest I found for foundry on that https://github.com/hbarcelos/forge-multi-version

**Comment by zerosnacks** [comment]
Marking as `out of scope` as the request specifically relates to having two different solc versions being applied to the same file which relates to the compiler and the request is better handled there.

Workarounds to test against multiple solc versions is possible by running `forge test` in parallel in combination with specific profiles

https://github.com/Vectorized/solady/blob/43f9d49815c8126d92771b26bd9bdbe2dbea87a5/.github/workflows/ci.yml#L34-L79

Alternatively you can use the `vm.etch` with the runtime code or https://book.getfoundry.sh/reference/forge-std/deployCode with a path to the compiled artifact
